qt/python/mantidqtinterfaces/mantidqtinterfaces/Engineering/gui/engineering_diffraction/tabs/gsas2/call_G2sc.py

```diff
@@ -0,0 +1,171 @@
+# Mantid Repository : https://github.com/mantidproject/mantid
+#
+# Copyright &copy; 2022 ISIS Rutherford Appleton Laboratory UKRI,
+#   NScD Oak Ridge National Laboratory, European Spallation Source,
+#   Institut Laue - Langevin & CSNS, Institute of High Energy Physics, CAS
+# SPDX - License - Identifier: GPL - 3.0 +
+import os
+import sys
+import json
+import numpy as np
+
+
+'''Parse Inputs from Mantid'''
+inputs_json = sys.argv[1]
+inputs_dict = json.loads(inputs_json)
+
+path_to_gsas2 = inputs_dict['path_to_gsas2']
+temporary_save_directory = inputs_dict['temporary_save_directory']
+data_directory = inputs_dict['data_directory']
+refinement_method = inputs_dict['refinement_method']
+project_name = inputs_dict['project_name']
+
+refine_background = inputs_dict['refine_background']
+refine_microstrain = inputs_dict['refine_microstrain']
+refine_sigma_one = inputs_dict['refine_sigma_one']
+refine_gamma = inputs_dict['refine_gamma']
+refine_histogram_scale_factor = inputs_dict['refine_histogram_scale_factor']
+
+data_files = inputs_dict['data_files']
+histogram_indexing = inputs_dict['histogram_indexing']
+
+# Check the number of input histograms
+number_histograms = len(data_files)
+if histogram_indexing and len(data_files) == 1:
+    number_histograms = len(histogram_indexing)
+# if histogram_indexing and len(data_files) > 1 should be caught in Validation
+
+phase_files = inputs_dict['phase_files']
+instrument_files = inputs_dict['instrument_files']
+x_min, x_max = inputs_dict['limits']
+mantid_pawley_reflections = inputs_dict['mantid_pawley_reflections']
+override_cell_lengths = inputs_dict['override_cell_lengths']
+
+
+'''Call GSASIIscriptable'''
+project_path = os.path.join(temporary_save_directory, project_name + '.gpx')
+
+sys.path.insert(0, path_to_gsas2 + 'GSASII')
+import GSASIIscriptable as G2sc  # noqa: E402
```

Comment:
Add a try block??? First see what the error output is when this fails, cos maybe it failing as is gives good feedback to the user?